horizomer/utils/tree.py

```diff
@@ -0,0 +1,132 @@
+#!/usr/bin/env python
+
+# ----------------------------------------------------------------------------
+# Copyright (c) 2015--, The Horizomer Development Team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# ----------------------------------------------------------------------------
+
+#
+# functions relevant to phylogenetic tree operations
+#
+
+from skbio import TreeNode
+
+
+def compare_topology(node1, node2):
+    """Test whether the topologies of two trees are identical.
+
+    Parameters
+    ----------
+    tree1 : skbio.TreeNode
+        first tree in comparison
+    tree2 : skbio.TreeNode
+        second tree in comparison
+
+    Returns
+    -------
+    bool
+        whether the topologies are identical
+
+    Notes
+    -----
+    Flipping child nodes does not change topology.
+    Branch lengths are ignored.
+    """
+    if len(node1.children) != len(node2.children):
+        return False
+    elif len(node1.children) > 0:
+        # check if sorted child names are identical
+        children1 = sorted(node1.children, key=lambda x: x.name)
```

Comment:
what happens if a node name is None?

Reply:
Practically one cannot make a Newick tree with a taxon name being empty. One can only do `tree.fine('a').name = ''` or `None` in Python, and when there is `None` in a list, it cannot be sorted. But I think this exception can't happen in this context.

Reply:
I disagree: the Greengenes reference tree for SEPP contains a lot of internal nodes without names. Or another example: ```(((a:1,b:2):6,(c:3,d:4):7):8,e:5):9;```
I had a lot of trouble with those unnamed nodes :-/

Reply:
Please see below my long response to your long comment.